tools/serverpod_cli/lib/src/generator/open_api/objects/security.dart

```diff
@@ -0,0 +1,223 @@
+  /// the use of TLS.
+  final String openIdConnectUrl;
+  OpenIdSecurityScheme({
+    super.description,
+    required this.openIdConnectUrl,
+  });
+
+  @override
+  Map<String, dynamic> toJson() {
+    Map<String, dynamic> map = {};
+    map['type'] = SecuritySchemeType.openIdConnect.name;
+    if (description != null) map['description'] = description!;
+    map['openIdConnectUrl'] = openIdConnectUrl;
+    return map;
+  }
+}
+
+/// Allows configuration of the supported OAuth Flows.
+class OauthFlowObject {
+  /// Fixed Fields [implicit, password, clientCredentials, authorizationCode]
+  final OauthFlowField fieldName;
+
+  /// The authorization URL to be used for this flow
+  /// Applies to oauth2 ("implicit", "authorizationCode")
+  final Uri? authorizationUrl;
+
+  /// The token URL to be used for this flow.
+  /// Applies to oauth2 ("password", "clientCredentials", "authorizationCode")
+  final Uri? tokenUrl;
+
+  /// The URL to be used for obtaining refresh tokens
+  final Uri refreshUrl;
+
+  /// The available scopes for the OAuth2 security scheme. A map between
+  /// the scope name and a short description for it.
+  final Map<String, String> scopes;
+
+  /// The extensions properties are implemented as patterned fields that are
+  /// always prefixed by "x-".
+  final Map<String, dynamic> specificationExtensions;
+  OauthFlowObject({
+    required this.fieldName,
+    this.authorizationUrl,
+    this.tokenUrl,
+    required this.refreshUrl,
+    required this.scopes,
+    required this.specificationExtensions,
+  });
+  Map<String, dynamic> toJson() {
+    var map = <String, dynamic>{};
+    map[fieldName.name] = {};
+    if (tokenUrl != null) {
+      map[fieldName.name] = tokenUrl.toString();
+    }
+    if (authorizationUrl != null) {
+      map[fieldName.name] = authorizationUrl.toString();
+    }
+
+    map[fieldName.name] = refreshUrl.toString();
+    map[fieldName.name] = scopes;
+    if (specificationExtensions.isEmpty) {
+      for (var ext in specificationExtensions.entries) {
+        if (ext.key.startsWith('x-')) {
+          map[fieldName.name][ext.key] = ext.value;
+        }
+      }
+    }
+    return {};
+  }
+}
+
+enum OauthFlowField {
+  implicit,
+  password,
+  clientCredentials,
+  authorizationCode,
+}
+
+class SecurityRequirementObject {
+  final String name;
+  final SecuritySchemeObject securitySchemes;
+  SecurityRequirementObject({
+    required this.name,
+    required this.securitySchemes,
+  });
+  Map<String, dynamic> toJson([bool ref = false]) {
+    Map<String, dynamic> map = {};
+    map[name] = ref
+        ? _getScopeFromSecurityScheme(securitySchemes)
+        : securitySchemes.toJson();
+    return map;
+  }
+
+  dynamic _getScopeFromSecurityScheme(SecuritySchemeObject securitySchemes) {
+    if (securitySchemes is OauthSecurityScheme) {
+      /// eg  - googleOAuth: ['openid', 'profile']
+      return securitySchemes.flows.first.scopes.keys.toList();
+    }
+    return [];
+  }
+}
+
+/// WIP
```

Comment:
It would be much better to add a TODO comment here as they can be found throug VSCode to indentify things that need to be fixed for the future. E.g.,

`// TODO: Describe what needs to be fixed.`